src/backend/commands/createas.c

```diff
@@ -1808,3 +1820,38 @@ get_primary_key_attnos_from_query(Query *query, List **constraintList)
 
 	return keys;
 }
+
+/* 
+ * Create auto-refresh task for Dynamic Tables.
+ */
+static void
+create_dynamic_table_auto_refresh_task(ParseState *pstate, Relation DynamicTableRel, char *schedule)
+{
+	ObjectAddress refaddr;
+	ObjectAddress address;
+	StringInfoData buf;
+	char *dtname = NULL;
+
+	if (schedule == NULL)
+		schedule = DYNAMIC_TABLE_DEFAULT_REFRESH_INTERVAL;
+
+	/* Create auto refresh task. */
+	CreateTaskStmt *task_stmt = makeNode(CreateTaskStmt);
+
+	initStringInfo(&buf);
+	appendStringInfo(&buf, "gp_dynamic_table_refresh_%u", RelationGetRelid(DynamicTableRel));
+	task_stmt->taskname = pstrdup(buf.data);
+	task_stmt->schedule = pstrdup(schedule);
+	task_stmt->if_not_exists = false; /* report error if failed. */
+	dtname = quote_qualified_identifier(get_namespace_name(RelationGetNamespace(DynamicTableRel)),
+											 RelationGetRelationName(DynamicTableRel));
+	resetStringInfo(&buf);
+	appendStringInfo(&buf, "REFRESH DYNAMIC TABLE %s", dtname);
```

Comment:
if drop task manually,  drop dynamic table maybe failed, you can add this test case?

Reply:
Good catch!

> if drop task manually

We shouldn't allow that as it's part of DYNAMIC TABLE, I provide a job prefix `gp_dynamic_table_refresh` prefix to tell the users it's a internal one. In GPDB, users can't manage `gp` or `pg` prefix objects in principle.
However, a prefix is not enough now.

Reply:
```c++
DropTask(ParseState *pstate, DropTaskStmt *stmt)
{
    ObjectAddress address = InvalidObjectAddress;
    char          *username;
    Oid           jobid = InvalidOid;

    /* current username */
    username = GetUserNameFromId(GetUserId(), false);

    /* delete from pg_task */
    jobid = UnscheduleCronJob(stmt->taskname, username, InvalidOid, stmt->missing_ok);

    /* delete from pg_task_run_history according to the jobid */
    if (OidIsValid(jobid))
    {
        RemoveTaskRunHistoryByJobId(jobid);
        ObjectAddressSet(address, TaskRelationId, jobid);
        /* Clean up dependencies */
        deleteSharedDependencyRecordsFor(TaskRelationId, jobid, 0);
       +deleteDependencyRecordsFor(TaskRelationId, jobid, false);
    }

    return address;
}
```
try remove dependency  use `deleteDependencyRecordsFor`.

Reply:
> We shouldn't allow that as it's part of DYNAMIC TABLE,

I think a proper fix is forbid user dropping the task goes hand with Dynamic Table, else it volatiles our design:
> Dynamic Table is a an auto-refreshing materialized view

Reply:
Done, also forbid users to create task using our dynamic table prefix.